File: solutions/difference-of-squares/Swift/Sources/DifferenceOfSquares/DifferenceOfSquares.swift
class Squares {
    let number: Int

    init(_ number: Int) {
        self.number = number
    }

    var squareOfSum: Int {
        let sum = (1...number).reduce(0, +)
        return sum * sum
    }

    var sumOfSquares: Int {
        return (1...number).reduce(0) { $0 + ($1 * $1) }
    }

    var differenceOfSquares: Int {
        return squareOfSum - sumOfSquares
    }
}

File: solutions/difference-of-squares/Clojure/src/difference_of_squares.clj
(ns difference-of-squares)

(defn sum-of-squares [n]
  (* n (+ n 1) (+ 1 (* 2 n)) (/ 6))
)

(defn square-of-sum [n]
  (* n n (+ 1 n) (+ 1 n) (/ 4))
)

(defn difference [n]
  (- (square-of-sum n) (sum-of-squares n))
)
